test/Core.Test/Services/OrganizationServiceTests.cs

```diff
@@ -138,5 +146,216 @@ public async Task OrgImportCreateNewUsersAndMarryExistingUser()
             await orgUserRepo.Received(1).UpsertAsync(Arg.Any<OrganizationUser>());
             await orgUserRepo.Received(2).CreateAsync(Arg.Any<OrganizationUser>());
         }
+
+        [Theory, CustomAutoDataAttribute(typeof(SutProviderCustomization))]
+        public async Task UpgradePlan_OrganizationIsNull_Throws(Guid organizationId, OrganizationUpgrade upgrade,
+                SutProvider<OrganizationService> sutProvider)
+        {
+            sutProvider.GetDependency<IOrganizationRepository>().GetByIdAsync(organizationId).Returns(Task.FromResult<Organization>(null));
+            var exception = await Assert.ThrowsAsync<NotFoundException>(
+                () => sutProvider.Sut.UpgradePlanAsync(organizationId, upgrade));
+        }
+
+        [Theory, CustomAutoDataAttribute(typeof(SutProviderCustomization))]
+        public async Task UpgradePlan_GatewayCustomIdIsNull_Throws(Organization organization, OrganizationUpgrade upgrade,
+                SutProvider<OrganizationService> sutProvider)
+        {
+            organization.GatewayCustomerId = string.Empty;
```

Comment:
Test name is that it's testing null, but this tests empty. Why not test both and send in the gatewayId as a paremeter?

Reply:
The check in the code is a NullOrWhitespace(). I know the tests should assume as little as possible about the code being tested, but does checking both in that context seem worth making a more complex test? In this case that would mean creating a whole new autodata attribute, wouldn't it?

Reply:
No, you could use InlineAutoData, unless that doesn't exist for organizations, but really I think this test is fine